openmls/src/group/tests/test_proposal_validation.rs

```diff
@@ -753,25 +865,46 @@ fn test_valsem104(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
     for alice_and_bob_share_keys in [
         KeyUniqueness::NegativeSameKey,
         KeyUniqueness::PositiveDifferentKey,
+        KeyUniqueness::PositiveSameKeyWithRemove,
     ] {
         // 0. Initialize Alice and Bob
-        let (alice_signature_keypair, bob_signature_keypair) = match alice_and_bob_share_keys {
-            KeyUniqueness::NegativeSameKey => {
-                let shared_signature_keypair =
+        let (alice_signature_keypair, bob_signature_keypair, target_signature_keypair) =
```

Comment:
Super hard to suggest something since diff is too big but please init credential bundle this way (btw you can rename ~credential_bundle~ into ~cb~ to have something that fits within a screen)

```rust
let new_kp = || SignatureKeypair::new(ciphersuite.signature_algorithm(), backend).unwrap();
let kp = new_kp();
let [alice_credential_bundle, bob_credential_bundle, target_credential_bundle] =
    match alice_and_bob_share_keys {
        KeyUniqueness::PositiveDifferentKey => [("Alice", kp.clone()), ("Bob", new_kp()), ("", kp)],
        KeyUniqueness::NegativeSameKey => [("Alice", new_kp()), ("Bob", new_kp()), ("Charlie", new_kp()), ],
        KeyUniqueness::PositiveSameKeyWithRemove => [("Alice", new_kp()), ("Bob", kp.clone()), ("Charlie", kp)],
    }
    .map(|(id, k)| CredentialBundle::from_parts(id.into(), k));
```